### PR TITLE
feat(connector): [NOVALNET] Add template code

### PR DESCRIPTION
**Pull request by dg-212**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Template code added for new connector Novalnet
https://developer.novalnet.com/onlinepayments/api

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/api_models/src/enums.rs [hunk 1]
```diff
@@ -46,6 +46,7 @@ pub enum RoutingAlgorithm {
 #[serde(rename_all = "snake_case")]
 #[strum(serialize_all = "snake_case")]
 pub enum Connector {
+	Novalnet,
```
Comment (deepanshu-iiitu) [inline crates/api_models/src/enums.rs:49]: Comment this line

File: crates/api_models/src/enums.rs [hunk 2]
```diff
@@ -209,6 +210,7 @@ impl Connector {
             | Self::DummyConnector7 => false,
             Self::Aci
             // Add Separate authentication support for connectors
+			| Self::Novalnet
```
Comment (deepanshu-iiitu) [inline crates/api_models/src/enums.rs:213]: Comment this line

File: crates/hyperswitch_connectors/src/default_implementations.rs [hunk 3]
```diff
@@ -86,6 +86,7 @@ macro_rules! default_imp_for_authorize_session_token {
 }
 
 default_imp_for_authorize_session_token!(
+	connectors::Novalnet,
```
Comment (deepanshu-iiitu) [inline crates/hyperswitch_connectors/src/default_implementations.rs:89]: Please maintain the alphabetical order in all places of this file.

File: crates/hyperswitch_connectors/src/default_implementations_v2.rs [hunk 4]
```diff
@@ -181,6 +181,7 @@ macro_rules! default_imp_for_new_connector_integration_payment {
 }
 
 default_imp_for_new_connector_integration_payment!(
+	connectors::Novalnet,
```
Comment (deepanshu-iiitu) [inline crates/hyperswitch_connectors/src/default_implementations_v2.rs:184]: Please maintain the alphabetical order in all places of this file.

File: crates/router/src/core/admin.rs [hunk 5]
```diff
@@ -1542,6 +1542,7 @@ impl<'a> ConnectorAuthTypeAndMetadataValidation<'a> {
                 threedsecureio::transformers::ThreedsecureioAuthType::try_from(self.auth_type)?;
                 Ok(())
             }
+            api_enums::Connector::Novalnet => todo!(),
```
Comment (deepanshu-iiitu) [inline crates/router/src/core/admin.rs:1545]: Comment this line

File: crates/router/src/types/api.rs [hunk 6]
```diff
@@ -497,6 +498,81 @@ impl ConnectorData {
...
+                api_models::enums::Connector::Volt => todo!(),
+                api_models::enums::Connector::Wellsfargo => todo!(),
+                api_models::enums::Connector::Wise => todo!(),
+                api_models::enums::Connector::Worldline => todo!(),
+                api_models::enums::Connector::Worldpay => todo!(),
+                api_models::enums::Connector::Signifyd => todo!(),
+                api_models::enums::Connector::Plaid => todo!(),
+                api_models::enums::Connector::Riskified => todo!(),
+                api_models::enums::Connector::Zen => todo!(),
+                api_models::enums::Connector::Zsl => todo!(),
```
Comment (deepanshu-iiitu) [inline crates/router/src/types/api.rs:575]: Remove this block

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

File: crates/router/src/types/api.rs [hunk 7]
```diff
@@ -497,6 +498,81 @@ impl ConnectorData {
...
```
Comment (deepanshu-iiitu) [inline crates/router/src/types/api.rs:575]: This is not needed. Please remove this.

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**